DatadogCore/Sources/Core/Storage/FilesOrchestrator.swift

```diff
@@ -249,7 +251,7 @@ internal class FilesOrchestrator: FilesOrchestratorType {
                 BatchDeletedMetric.uploaderWindowKey: performance.uploaderWindow.toMilliseconds,
                 BatchDeletedMetric.batchAgeKey: batchAge.toMilliseconds,
                 BatchDeletedMetric.batchRemovalReasonKey: deletionReason.toString(),
-                BatchDeletedMetric.inBackgroundKey: false
+                BatchDeletedMetric.inBackgroundKey: metricsData.backgroundTasksEnabled
```

Comment:
I understand you want to use an existing, unused key, but isn't it a bit confusing to mix them? Can't we create a `BatchDeletedMetric.backgroundTasksEnabledKey`?

Reply:
Agree with @mariedm, let's create a new key and add it to the [spec](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3287975594/Batch+Deleted+Telemetry+-+Spec). The `in_background` has different purpose:
> `in_background` - if the batch was deleted in background; true or false

Having `backgroundTasksEnabled` enabled doesn't imply that all batches are deleted in background, so we set this value wrong. It was originally hardcoded to `false`, because iOS SDK was not supporting background uploads.

Supporting the actual value in `in_background` is separate task that will require complex implementation.